src/mpyl/utilities/docker/__init__.py

```diff
@@ -338,3 +350,72 @@ def remove_container(logger: Logger, container: Container) -> None:
     logger.debug(f"Removing container {container.id}")
     docker.remove(container.id)
     logger.info(f"Removed container {container.id}")
+
+
+def create_ecr_repo_if_needed(
+    logger: Logger, registry_config: DockerRegistryConfig, repo: str
+):
+    ecr_config = Config(
+        region_name=registry_config.region,
+        signature_version="v4",
+        retries={"max_attempts": 10, "mode": "standard"},
+    )
+    ecr_client = boto3.client(
+        "ecr",
+        config=ecr_config,
+        aws_access_key_id=registry_config.user_name,
+        aws_secret_access_key=registry_config.password,
+    )
+    try:
+        ecr_client.describe_repositories(
+            repositoryNames=[
+                repo.lower(),
+            ]
+        )
+        logger.info(f"Repository '{repo}' exists.")
+    except (
+        ecr_client.exceptions.InvalidParameterException,
+        ecr_client.exceptions.ServerException,
+    ) as exc:
+        raise exc
+    except ecr_client.exceptions.RepositoryNotFoundException:
+        logger.info(f"Repository '{repo}' not found. Creating...")
+        try:
+            ecr_client.create_repository(
+                repositoryName=repo.lower(),
+                imageTagMutability="IMMUTABLE",
+                encryptionConfiguration={
+                    "encryptionType": "AES256",
+                },
+            )
+        except (
+            ecr_client.exceptions.InvalidTagException,
+            ecr_client.exceptions.TooManyTagsException,
+            ecr_client.exceptions.LimitExceededException,
+        ) as exc:
+            raise exc
+        ecr_lifecycle_policy(logger, ecr_client, repo)
+    logger.info(f"ECR ready, pushing image to {repo}...")
+
+
+def ecr_lifecycle_policy(logger: Logger, ecr_client, repo):
+    lifecycle_policy = {
```

Comment:
Did you and Daniel discuss if this should be in mpyl or the config? Basically, if this might need changes or could be repo/company specific, then it's better in the config.

Reply:
good one. I'll add it to the config